roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -97,43 +97,37 @@ impl JDsMempool {
 
     pub async fn update_mempool(self_: Arc<Mutex<Self>>) -> Result<(), JdsMempoolError> {
         let mut mempool_ordered: HashMap<Txid, Option<Transaction>> = HashMap::new();
+
         let client = self_
             .safe_lock(|x| x.get_client())
             .map_err(|e| JdsMempoolError::PoisonLock(e.to_string()))?
             .ok_or(JdsMempoolError::NoClient)?;
-        let new_mempool: Result<HashMap<Txid, Option<Transaction>>, JdsMempoolError> = {
-            let self_ = self_.clone();
-            tokio::task::spawn(async move {
-                let mempool: Vec<String> = client
-                    .get_raw_mempool()
-                    .await
-                    .map_err(JdsMempoolError::Rpc)?;
-                for id in &mempool {
-                    let key_id = Txid::from_str(id).unwrap();
-                    let tx = self_.safe_lock(|x| match x.mempool.get(&key_id) {
-                        Some(entry) => entry.clone(),
-                        None => None,
-                    });
-                    let id = Txid::from_str(id).unwrap();
-                    mempool_ordered.insert(id, tx.unwrap());
-                }
-                if mempool_ordered.is_empty() {
-                    Err(JdsMempoolError::EmptyMempool)
-                } else {
-                    Ok(mempool_ordered)
-                }
-            })
-            .await
-            .map_err(JdsMempoolError::TokioJoin)?
-        };
-        match new_mempool {
-            Ok(new_mempool_) => {
-                let _ = self_.safe_lock(|x| {
-                    x.mempool = new_mempool_;
-                });
-                Ok(())
-            }
-            Err(a) => Err(a),
+
+        let mempool: Vec<String> = client.get_raw_mempool().await.map_err(|err| {
+            let err_msg = format!("Error occurred during deserialization: {:?}", err);
+            JdsMempoolError::Rpc(RpcError::Deserialization(err_msg))
+        })?;
+        for id in &mempool {
+            let key_id = Txid::from_str(id).map_err(|err| {
+                let err_msg = format!("Error occurred during deserialization: {:?}", err);
```

Comment:
Also this mapping seems problematic. Is true that here you can have only de-serialization error. But my question is what this mapping add to the error? Why not just immediately return the error without mapping it?

Reply:
I spoke to @lorbax ftom our previous conversation on this PR and he said I should return a deserialization error

Reply:
If there is not TryFrom for it would be better to add it but is ok also like that, for the other 2 map you need to remove the mapping

Reply:
BTW for sure the most elegant solution is to implement from for error conversion as @Fi3 already suggested